Add `orNull` functions for non-throwing construction of datetime entities

dkhalanskyjb opened the pull request:
Fixes #68

dkhalanskyjb committed:
Add LocalDate.createOrNull

dkhalanskyjb committed:
Generate `createOrNull` implementations using AI

Prompts:
"""
In the last commit, `LocalDate.createOrNull` was introduced,
along with a usage sample, and with the existing tests modified
to also check `createOrNull`.
You should introduce `LocalTime.createOrNull` in the same way.
***
Now, similarly, introduce `LocalDateTime.createOrNull`
with tests and a usage sample.
***
Now also introduce `UtcOffset.createOrNull`
(with tests and a usage sample), except that since the constructor
for `UtcOffset` is internal, use the behavior from `fun UtcOffset`.
"""

dkhalanskyjb committed:
Fix the output of AI

dkhalanskyjb committed:
Update the docs

dkhalanskyjb committed:
Rename createOrNull to orNull

dkhalanskyjb committed:
Avoid extraneous range checks when constructors receive safe values

dkhalanskyjb committed:
Cleanup

ilya-g commented on core/common/test/LocalDateTimeTest.kt:
```diff
@@ -131,6 +131,40 @@ class LocalDateTimeTest {
     }
 
+    @Test
+    fun orNull() {// Test orNull with month number
```
Minor: comment on a new line

dkhalanskyjb committed:
Fix after a rebase

dkhalanskyjb committed:
Add `orNull` to `YearMonth` and `LocalIsoWeekDate`

Implemented by an LLM; prompt:
"Add orNull to YearMonth and LocalIsoWeekDate.
Use git diff from origin/master for inspiration.
Add code, docs, tests, samples."

dkhalanskyjb committed:
Cleanup after the LLM
